### ISSUE TITLE
Running serverside

### ISSUE THREAD

**Issue by edorivai** [mixpanel/mixpanel-js#119]
Hi!

I know this module is not meant to be ran serverside, and I don't plan to actually track stuff serverside through this module. Thing is, I have this universal app, which renders serverside. Now running code which imports `mixpanel-browser` crashes my node server, understandably complaining that `document` is not defined:

```
var HTTP_PROTOCOL = (('https:' === document.location.protocol) ? 'https://' : 'http://');
                                   ^
ReferenceError: document is not defined
```

It would work for me if you could just short-circuit the whole module as soon as you detect that the document is not defined, instead of crashing the whole server. Currently I'm using a custom wrapper like so:

```js
const mixpanel = __CLIENT__ ? require('mixpanel-browser') : { track: () => {} };

export { mixpanel as default };
```

It does the job, but I'm wondering whether catching this within the module might be more elegant :)

**Comment by tdumitrescu** [comment]
There are actually more browser-specific dependencies than just `document` in the lib, and we're not super-keen on putting defensive code around it all. See the discussion in https://github.com/mixpanel/mixpanel-js/pull/102. With regard to short-circuiting the module-loading entirely, I think your existing solution is fine for a CommonJS environment and if you want to open a PR with similar functionality go for it; but we build the lib from ES2015 imports which should be treated as static, i.e. don't rely on a module getting conditionally imported the way you can with CJS `require`s.

**Comment by edorivai** [comment]
Makes sense! Thanks for the hints, I'll take a closer look on Monday, might open a PR then ;)

**Comment by edorivai** [comment]
Hey again,

I tried polyfilling the global properties, as suggested [here](https://github.com/mixpanel/mixpanel-js/pull/102#issuecomment-241127549). Problem is, _a lot_ of libraries check for a non-browser environment by testing whether `typeof document === 'undefined'`. So in this case it's not possible for me to simply polyfill them.

As you said, my current fix works for a CommonJS environment (which we're technically running right now), but I'd prefer the static ES2015 imports. So given static imports, and polyfilling not being an option, the only options seems to be fixing it in the base library code. I understand you don't want to write defensive code around each of the browser features, so I see two options:

1. Use a local polyfill, like you're already doing in `utils.js` right now. This way the polyfill doesn't touch  the global namespace, and you have one place where you declare your browser specific properties.
2. Shortcircuit the entire library, and return a mock if the DOM features are not present. This would involve a single check, but it would also require the DOM specific code to be wrapped in a function, so you could prevent it from executing. This would obviously also make the library useless in a server environment, but in my case at least, this is a non-issue.

Please let me know what you think!
